assets/chat/js/emotes.js

```diff
@@ -1,42 +1,59 @@
 export default class EmoteService {
-  emotes = [];
+  emotes = new Map();
```

Comment:
Is storing emotes in a map necessarily better than an array? It means calls to `canUserUseEmote()` are "more efficient", but negatively affects every other method because you have to get the map's values, first.

I think I'd prefer if you left `emotes` an array. If you're interested in improving the performance of this service in a future PR, you can instead preprocess the emotes such that you don't have to perform "expensive" filter or find operations. This may involve using a combination of a map and sorted arrays that contain pre-filtered sets of emotes. I use "expensive" in quotes because I don't think this is a performance pain point to begin with, but not 100% sure.